build main branch on m1 failed

```
+ ./configure --with-internal-glib --prefix=/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig
checking for a BSD-compatible install... /opt/homebrew/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /opt/homebrew/bin/gmkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... arm-apple-darwin23.4.0
checking host system type... arm-apple-darwin23.4.0
checking how to print strings... printf
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for a sed that does not truncate output... /opt/homebrew/bin/gsed
checking for grep that handles long lines and -e... /opt/homebrew/bin/ggrep
checking for egrep... /opt/homebrew/bin/ggrep -E
checking for fgrep... /opt/homebrew/bin/ggrep -F
checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 786432
checking how to convert arm-apple-darwin23.4.0 file names to arm-apple-darwin23.4.0 format... func_convert_file_noop
checking how to convert arm-apple-darwin23.4.0 file names to toolchain format... func_convert_file_noop
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... ld: warning: -single_module is obsolete
no
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin23.4.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking whether make supports nested variables... (cached) yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking for ln... ln
checking dirent.h usability... yes
checking dirent.h presence... yes
checking for dirent.h... yes
checking for unistd.h... (cached) yes
checking sys/wait.h usability... yes
checking sys/wait.h presence... yes
checking for sys/wait.h... yes
checking malloc.h usability... no
checking malloc.h presence... no
checking for malloc.h... no
checking for bash... /opt/homebrew/bin/bash
checking for default search path for .pc files... ${libdir}/pkgconfig:${datadir}/pkgconfig
checking for system include path to avoid -I flags... /usr/include
checking for system library path to avoid -L flags... /usr/lib:/lib
configure: creating ./config.lt
config.lt: creating libtool
checking whether to list both direct and indirect dependencies... no
checking for Win32... no
checking if prefix should be redefined at runtime... no
checking if internal glib should be used... yes
checking if host- prefixed tool should be installed... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating pkg.m4
config.status: creating check/Makefile
config.status: creating check/config.sh
config.status: creating config.h.win32
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
=== configuring in glib (/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig'  '--with-internal-glib' --cache-file=/dev/null --srcdir=.
checking for a BSD-compatible install... /opt/homebrew/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /opt/homebrew/bin/gmkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '501' is supported by ustar format... yes
checking whether GID '20' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /opt/homebrew/bin/ggrep
checking for egrep... /opt/homebrew/bin/ggrep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking build system type... arm-apple-darwin23.4.0
checking host system type... arm-apple-darwin23.4.0
checking for the BeOS... no
checking for Win32... no
checking for the Android... no
checking for Mac OS X Carbon support... yes
checking for Mac OS X Cocoa support... yes
checking whether to enable garbage collector friendliness... no
checking whether to disable memory pools... no
checking for c++... c++
checking whether we are using the GNU C++ compiler... yes
checking whether c++ accepts -g... yes
checking dependency style of c++... gcc3
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for gawk... (cached) gawk
checking for perl5... no
checking for perl... perl
checking for indent... indent
checking for perl... /usr/bin/perl
checking for a Python interpreter with version >= 2.5... python3
checking for python3... /usr/local/bin/python3
checking for python3 version... 3.1
checking for python3 platform... darwin
checking for python3 script directory... ${prefix}/lib/python3.1/site-packages
checking for python3 extension module directory... ${exec_prefix}/lib/python3.1/site-packages
checking for iconv_open... no
checking for libiconv_open in -liconv... no
checking for iconv_open in -liconv... yes
checking how to print strings... printf
checking for a sed that does not truncate output... /opt/homebrew/bin/gsed
checking for fgrep... /opt/homebrew/bin/ggrep -F
checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm -B
checking the name lister (/usr/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 786432
checking how to convert arm-apple-darwin23.4.0 file names to arm-apple-darwin23.4.0 format... func_convert_file_noop
checking how to convert arm-apple-darwin23.4.0 file names to toolchain format... func_convert_file_noop
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... no
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm -B output from gcc object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... ld: warning: -single_module is obsolete
no
checking for -exported_symbols_list linker flag... yes
checking for -force_load linker flag... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin23.4.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking how to run the C++ preprocessor... c++ -E
checking for ld used by c++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking whether the c++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking for c++ option to produce PIC... -fno-common -DPIC
checking if c++ PIC flag -fno-common -DPIC works... yes
checking if c++ static flag -static works... no
checking if c++ supports -c -o file.o... yes
checking if c++ supports -c -o file.o... (cached) yes
checking whether the c++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin23.4.0 dyld
checking how to hardcode library paths into programs... immediate
configure: creating ./config.lt
config.lt: creating libtool
checking for extra flags to get ANSI library prototypes... none needed
checking for extra flags for POSIX compliance... none needed
checking for vprintf... yes
checking for _doprnt... no
checking for size_t... yes
checking for working alloca.h... yes
checking for alloca... yes
checking for mmap... yes
checking for posix_memalign... yes
checking for memalign... no
checking for valloc... yes
checking for fsync... yes
checking for pipe2... no
checking for issetugid... yes
checking for atexit... yes
checking for on_exit... no
checking for timegm... yes
checking for gmtime_r... yes
checking for __libc_enable_secure... no
checking size of char... 1
checking size of short... 2
checking size of long... 8
checking size of int... 4
checking size of void *... 8
checking size of long long... 8
checking size of __int64... 0
checking for sig_atomic_t... yes
checking for format to printf and scanf a guint64... none
checking for an ANSI C-conforming const... yes
checking for growing stack pointer... yes
checking for __inline... yes
checking for __inline__... yes
checking for inline... yes
checking if inline functions in headers work... yes
checking for working do while(0) macros... no
checking for ISO C99 varargs macros in C... yes
checking for ISO C99 varargs macros in C++... yes
checking for GNUC varargs macros... yes
checking for GNUC visibility attribute... no
checking whether using Sun Studio C compiler... no
checking whether byte ordering is bigendian... no
checking dirent.h usability... yes
checking dirent.h presence... yes
checking for dirent.h... yes
checking float.h usability... yes
checking float.h presence... yes
checking for float.h... yes
checking limits.h usability... yes
checking limits.h presence... yes
checking for limits.h... yes
checking pwd.h usability... yes
checking pwd.h presence... yes
checking for pwd.h... yes
checking grp.h usability... yes
checking grp.h presence... yes
checking for grp.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking sys/poll.h usability... yes
checking sys/poll.h presence... yes
checking for sys/poll.h... yes
checking sys/resource.h usability... yes
checking sys/resource.h presence... yes
checking for sys/resource.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/times.h usability... yes
checking sys/times.h presence... yes
checking for sys/times.h... yes
checking sys/wait.h usability... yes
checking sys/wait.h presence... yes
checking for sys/wait.h... yes
checking for unistd.h... (cached) yes
checking values.h usability... no
checking values.h presence... no
checking for values.h... no
checking sys/select.h usability... yes
checking sys/select.h presence... yes
checking for sys/select.h... yes
checking for sys/types.h... (cached) yes
checking for stdint.h... (cached) yes
checking for inttypes.h... (cached) yes
checking sched.h usability... yes
checking sched.h presence... yes
checking for sched.h... yes
checking malloc.h usability... no
checking malloc.h presence... no
checking for malloc.h... no
checking sys/vfs.h usability... no
checking sys/vfs.h presence... no
checking for sys/vfs.h... no
checking sys/vmount.h usability... no
checking sys/vmount.h presence... no
checking for sys/vmount.h... no
checking sys/statfs.h usability... no
checking sys/statfs.h presence... no
checking for sys/statfs.h... no
checking sys/statvfs.h usability... yes
checking sys/statvfs.h presence... yes
checking for sys/statvfs.h... yes
checking sys/filio.h usability... yes
checking sys/filio.h presence... yes
checking for sys/filio.h... yes
checking mntent.h usability... no
checking mntent.h presence... no
checking for mntent.h... no
checking sys/mnttab.h usability... no
checking sys/mnttab.h presence... no
checking for sys/mnttab.h... no
checking sys/vfstab.h usability... no
checking sys/vfstab.h presence... no
checking for sys/vfstab.h... no
checking sys/mntctl.h usability... no
checking sys/mntctl.h presence... no
checking for sys/mntctl.h... no
checking fstab.h usability... yes
checking fstab.h presence... yes
checking for fstab.h... yes
checking sys/uio.h usability... yes
checking sys/uio.h presence... yes
checking for sys/uio.h... yes
checking sys/mkdev.h usability... no
checking sys/mkdev.h presence... no
checking for sys/mkdev.h... no
checking linux/magic.h usability... no
checking linux/magic.h presence... no
checking for linux/magic.h... no
checking sys/prctl.h usability... no
checking sys/prctl.h presence... no
checking for sys/prctl.h... no
checking for sys/mount.h... yes
checking for sys/sysctl.h... yes
checking for sysctlbyname... yes
checking xlocale.h usability... yes
checking xlocale.h presence... yes
checking for xlocale.h... yes
checking for struct stat.st_mtimensec... no
checking for struct stat.st_mtim.tv_nsec... no
checking for struct stat.st_atimensec... no
checking for struct stat.st_atim.tv_nsec... no
checking for struct stat.st_ctimensec... no
checking for struct stat.st_ctim.tv_nsec... no
checking for struct stat.st_blksize... yes
checking for struct stat.st_blocks... yes
checking for struct statfs.f_fstypename... yes
checking for struct statfs.f_bavail... yes
checking for struct statvfs.f_basetype... no
checking for struct statvfs.f_fstypename... no
checking for struct tm.tm_gmtoff... yes
checking for struct tm.__tm_gmtoff... no
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking for struct dirent.d_type... yes
checking for nl_langinfo and CODESET... yes
checking stddef.h usability... yes
checking stddef.h presence... yes
checking for stddef.h... yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for setlocale... yes
checking whether we are using the GNU C Library 2.1 or newer... no
checking size of size_t... 8
checking for the appropriate definition for size_t... unsigned long
checking size of ssize_t... 8
checking for the appropriate definition for ssize_t... long
checking for lstat... yes
checking for strerror... yes
checking for strsignal... yes
checking for memmove... yes
checking for vsnprintf... yes
checking for stpcpy... yes
checking for strcasecmp... yes
checking for strncasecmp... yes
checking for poll... yes
checking for getcwd... yes
checking for vasprintf... yes
checking for setenv... yes
checking for unsetenv... yes
checking for getc_unlocked... yes
checking for readlink... yes
checking for symlink... yes
checking for fdwalk... no
checking for memmem... yes
checking for chown... yes
checking for lchmod... yes
checking for lchown... yes
checking for fchmod... yes
checking for fchown... yes
checking for link... yes
checking for utimes... yes
checking for getgrgid... yes
checking for getpwuid... yes
checking for getresuid... no
checking for getmntent_r... no
checking for setmntent... no
checking for endmntent... no
checking for hasmntopt... no
checking for getfsstat... yes
checking for getvfsstat... no
checking for fallocate... no
checking for splice... no
checking for prlimit... no
checking for statvfs... yes
checking for statfs... yes
checking whether to use statfs or statvfs... statfs
checking crt_externs.h usability... yes
checking crt_externs.h presence... yes
checking for crt_externs.h... yes
checking for _NSGetEnviron... yes
checking for newlocale... yes
checking for uselocale... yes
checking for strtod_l... yes
checking for strtoll_l... yes
checking for strtoull_l... yes
checking for C99 vsnprintf... no
checking whether printf supports positional parameters... no
checking value of AF_INET... 2
checking value of AF_INET6... 30
checking value of AF_UNIX... 1
checking value of MSG_PEEK... 2
checking value of MSG_OOB... 1
checking value of MSG_DONTROUTE... 4
checking for getprotobyname_r... no
checking for endservent... yes
checking for if_nametoindex... yes
checking for if_indextoname... yes
checking if arpa/nameser_compat.h is needed... yes
checking for res_query... in -lresolv
checking for socket... yes
checking for res_init... yes
checking for linux/netlink.h... no
checking for struct ip_mreqn... yes
checking number of arguments to statfs()... 2
checking for signed... yes
checking for long long... yes
checking for long double... yes
checking for wchar_t... yes
checking for wint_t... yes
checking for size_t... (cached) yes
checking for ptrdiff_t... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for snprintf... yes
checking for wcslen... yes
checking for C99 snprintf... no
checking for fd_set... yes, found in sys/types.h
checking for nl_langinfo (CODESET)... yes
checking for nl_langinfo (PM_STR)... yes
checking for nl_langinfo (_NL_CTYPE_OUTDIGITn_MB)... no
checking for a compliant posix_memalign() implementation... yes
checking for OpenBSD strlcpy/strlcat... yes
checking for an implementation of va_copy()... yes
checking for an implementation of __va_copy()... yes
checking whether va_lists can be copied by value... yes
checking for dlopen... yes
checking for dlsym... yes
checking for RTLD_GLOBAL brokenness... no
checking for preceeding underscore in symbols... no
checking for dlerror... yes
checking for the suffix of module shared libraries... .so
checking for gspawn implementation... gspawn.lo
checking for GIOChannel implementation... giounix.lo
checking sys/inotify.h usability... no
checking sys/inotify.h presence... no
checking for sys/inotify.h... no
checking sys/event.h usability... yes
checking sys/event.h presence... yes
checking for sys/event.h... yes
checking for kqueue... yes
checking for kevent... yes
checking for platform-dependent source... 
checking whether to compile timeloop... yes
checking if building for some Win32 platform... no
checking for thread implementation... posix
checking thread related cflags... -D_REENTRANT
checking for pthread_create/pthread_join... yes
checking thread related libraries... 
checking for localtime_r... yes
checking for gmtime_r... (cached) yes
checking for posix getpwuid_r... yes
checking for posix getgrgid_r... yes
checking for pthread_attr_setstacksize... yes
checking for pthread_condattr_setclock... no
checking for pthread_cond_timedwait_monotonic... no
checking for pthread_cond_timedwait_monotonic_np... no
checking for clock_gettime... yes
checking for lock-free atomic intrinsics... yes
checking for futex(2) system call... no
checking for eventfd(2) system call... no
checking value of POLLIN... 1
checking value of POLLOUT... 4
checking value of POLLPRI... 2
checking value of POLLERR... 8
checking value of POLLHUP... 16
checking value of POLLNVAL... 32
checking for broken poll... yes
checking for EILSEQ... yes
checking for guint32... yes
checking alignment of guint32... 4
checking for guint64... yes
checking alignment of guint64... 8
checking for unsigned long... yes
checking alignment of unsigned long... 8
checking for -Bsymbolic-functions linker flag... no
checking for -fvisibility=hidden compiler flag... no
checking if gcc supports flag -Wall in envvar CFLAGS... yes
checking if gcc supports flag -Wstrict-prototypes in envvar CFLAGS... yes
checking if gcc supports flag -Werror=declaration-after-statement in envvar CFLAGS... yes
checking if gcc supports flag -Werror=missing-prototypes in envvar CFLAGS... yes
checking if gcc supports flag -Werror=implicit-function-declaration in envvar CFLAGS... yes
checking if gcc supports flag -Werror=pointer-arith in envvar CFLAGS... yes
checking if gcc supports flag -Werror=init-self in envvar CFLAGS... yes
checking if gcc supports flag -Werror=format-security in envvar CFLAGS... yes
checking if gcc supports flag -Werror=format=2 in envvar CFLAGS... yes
checking if gcc supports flag -Werror=missing-include-dirs in envvar CFLAGS... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating glib/Makefile
config.status: creating glib/libcharset/Makefile
config.status: creating glib/gnulib/Makefile
config.status: creating m4macros/Makefile
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing glib/glibconfig.h commands
+ /private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make
/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make  all-recursive
make[1]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir'
Making all in glib
make[2]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib'
/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make  all-recursive
make[3]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib'
Making all in .
make[4]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib'
make[4]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib'
Making all in m4macros
make[4]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/m4macros'
/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make  all-am
make[5]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/m4macros'
make[5]: Nothing to be done for 'all-am'.
make[5]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/m4macros'
make[4]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/m4macros'
Making all in glib
make[4]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib'
  GEN      glibconfig-stamp
config.status: executing glib/glibconfig.h commands
config.status: glib/glibconfig.h is unchanged
/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make  all-recursive
make[5]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib'
Making all in libcharset
make[6]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/libcharset'
/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make  all-am
make[7]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/libcharset'
  CC       libcharset_la-localcharset.lo
  CCLD     libcharset.la
  GEN      ref-add.sed
  GEN      ref-del.sed
  GEN      charset.alias
make[7]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/libcharset'
make[6]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/libcharset'
Making all in gnulib
make[6]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/gnulib'
/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make  all-am
make[7]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/gnulib'
  CC       libgnulib_la-asnprintf.lo
  CC       libgnulib_la-printf-args.lo
  CC       libgnulib_la-printf-parse.lo
  CC       libgnulib_la-vasnprintf.lo
  CC       libgnulib_la-printf.lo
  CCLD     libgnulib.la
make[7]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/gnulib'
make[6]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib/gnulib'
Making all in .
make[6]: Entering directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib'
  CC       libglib_2_0_la-gallocator.lo
  CC       libglib_2_0_la-gcache.lo
  CC       libglib_2_0_la-gcompletion.lo
  CC       libglib_2_0_la-grel.lo
  CC       libglib_2_0_la-gthread-deprecated.lo
  CC       libglib_2_0_la-garray.lo
  CC       libglib_2_0_la-gasyncqueue.lo
  CC       libglib_2_0_la-gatomic.lo
gatomic.c:392:10: error: incompatible integer to pointer conversion passing 'gssize' (aka 'long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_add ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:170:46: note: expanded from macro 'g_atomic_pointer_add'
    (gssize) __sync_fetch_and_add ((atomic), (val));                         \
                                             ^~~~~
gatomic.c:416:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_and ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:177:45: note: expanded from macro 'g_atomic_pointer_and'
    (gsize) __sync_fetch_and_and ((atomic), (val));                          \
                                            ^~~~~
gatomic.c:440:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_or ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:184:44: note: expanded from macro 'g_atomic_pointer_or'
    (gsize) __sync_fetch_and_or ((atomic), (val));                           \
                                           ^~~~~
gatomic.c:464:10: error: incompatible integer to pointer conversion passing 'gsize' (aka 'unsigned long') to parameter of type 'gpointer' (aka 'void *') [-Wint-conversion]
  return g_atomic_pointer_xor ((volatile gpointer *) atomic, val);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./gatomic.h:191:45: note: expanded from macro 'g_atomic_pointer_xor'
    (gsize) __sync_fetch_and_xor ((atomic), (val));                          \
                                            ^~~~~
4 errors generated.
make[6]: *** [Makefile:1317: libglib_2_0_la-gatomic.lo] Error 1
make[6]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib'
make[5]: *** [Makefile:1934: all-recursive] Error 1
make[5]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib'
make[4]: *** [Makefile:953: all] Error 2
make[4]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib/glib'
make[3]: *** [Makefile:1046: all-recursive] Error 1
make[3]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib'
make[2]: *** [Makefile:770: all] Error 2
make[2]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir/glib'
make[1]: *** [Makefile:707: all-recursive] Error 1
make[1]: Leaving directory '/private/var/tmp/_bazel_zirain/7fa84ac355d3510cee072b40f586889f/sandbox/darwin-sandbox/948/execroot/envoy/bazel-out/darwin_arm64-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/pkgconfig.build_tmpdir'
make: *** [Makefile:463: all] Error 2

```

cc @phlax

--- a/.bazelrc
+++ b/.bazelrc
@@ -148,6 +148,12 @@ build:macos --action_env=PATH=/opt/homebrew/bin:/opt/local/bin:/usr/local/bin:/u
 build:macos --host_action_env=PATH=/opt/homebrew/bin:/opt/local/bin:/usr/local/bin:/usr/bin:/bin
 build:macos --define tcmalloc=disabled
 
+# Workaround issue with pkgconfig, see https://github.com/envoyproxy/envoy/issues/33225
+build:macos --host_action_env=CXXFLAGS=-Wno-int-conversion
+build:macos --action_env=CXXFLAGS=-Wno-int-conversion
+build:macos --host_action_env=CFLAGS=-Wno-int-conversion
+build:macos --action_env=CFLAGS=-Wno-int-conversion
+
 # macOS ASAN/UBSAN
 build:macos-asan --config=asan
 # Workaround, see https://github.com/bazelbuild/bazel/issues/6932

